ansible/roles/raspi-config/templates/dhcpd5.service.j2

```diff
@@ -0,0 +1,15 @@
+[Unit]
+Description=dhcpcd on all interfaces
+Wants=network.target
+Before=network.target
+Restart=always
+
+[Service]his i
+Type=forking
+PIDFile=/run/dhcpcd.pid
+ExecStart=/usr/lib/dhcpcd5/dhcpcd -w -q -t 0 %I
+ExecStop=/sbin/dhcpcd -x
+
+[Install]
+WantedBy=multi-user.target
+Alias=dhcpcd5.service
```

Comment:
Actually I have this file on the raspberry pi, where `ExecStart` is different : 
```
[Unit]
Description=dhcpcd on all interfaces
Wants=network.target
Before=network.target

[Service]
Type=forking
PIDFile=/run/dhcpcd.pid
ExecStart=/usr/lib/dhcpcd5/dhcpcd -q -b
ExecStop=/sbin/dhcpcd -x

[Install]
WantedBy=multi-user.target
Alias=dhcpcd5.service
```

According to systemd documentation default timeout is **90s**, we should double it to 180s / 2 min : 
```
[](https://www.freedesktop.org/software/systemd/man/systemd.service.html#TimeoutStartSec=)TimeoutStartSec=

    Configures the time to wait for start-up. If a daemon service does not signal start-up completion within the configured time, the service will be considered failed and will be shut down again. The precise action depends on the TimeoutStartFailureMode= option. Takes a unit-less value in seconds, or a time span value such as "5min 20s". Pass "infinity" to disable the timeout logic. Defaults to DefaultTimeoutStartSec= from the manager configuration file, except when Type=oneshot is used, in which case the timeout is disabled by default (see [systemd-system.conf(5)](https://www.freedesktop.org/software/systemd/man/systemd-system.conf.html#)).

[...]

[](https://www.freedesktop.org/software/systemd/man/systemd-system.conf.html#DefaultTimeoutStartSec=)DefaultTimeoutStartSec=, DefaultTimeoutStopSec=, DefaultTimeoutAbortSec=, DefaultRestartSec=

    Configures the default timeouts for starting, stopping and aborting of units, as well as the default time to sleep between automatic restarts of units, as configured per-unit in TimeoutStartSec=, TimeoutStopSec=, TimeoutAbortSec= and RestartSec= (for services, see [systemd.service(5)](https://www.freedesktop.org/software/systemd/man/systemd.service.html#) for details on the per-unit settings). Disabled by default, when service with Type=oneshot is used. For non-service units, DefaultTimeoutStartSec= sets the default TimeoutSec= value. DefaultTimeoutStartSec= and DefaultTimeoutStopSec= default to 90s. DefaultTimeoutAbortSec= is not set by default so that all units fall back to TimeoutStopSec=. DefaultRestartSec= defaults to 100ms.
```

We need to add, in `[Service]` section :
```
TimeoutStartSec=180
```

Reply:
I'm just increasing systemd timeout as the issue comes from systemd (killing dhcpd) not from dhcpd, I hope i will be sufficient.